### PR TITLE
Update deployment guide for different environments

### PR DESCRIPTION
**Pull request by dinoboy197**
Updated deployment instructions per updated install on 2024-04-27.

[Fully rendered markdown](https://github.com/CodeWithAloha/HIERR/blob/traack-deploy-updates/src/docs/deployment.md).

### CODE REVIEW

**Review by tyliec — APPROVED** [review]

File: src/docs/deployment.md [hunk 1]
```diff
@@ -1,142 +1,149 @@
...
+         1. A dialog box will open:
+            1. Type {HTTPS} in the Condition input field.
+            1. Set Check if input string to Matches the Pattern.
+            1. Type ^OFF$ in the Pattern field.
+            1. Check Ignore case.
+         1. Click the OK button.
+         1. You should now see your condition in the list.
+         1. Scroll down to the Action section and enter these settings:
+            1. Select Redirect as the Action type.
+            1. Type https://<Replace with your host name>/ in the Redirect URL field.
```
Comment (tyliec) [inline src/docs/deployment.md:127]: nit: This is showing up as this: 

Screenshot 2024-05-08 at 10 35 55 PM

File: src/docs/deployment.md [hunk 2]
```diff
@@ -1,142 +1,149 @@
...
+            1. Uncheck Append query string.
+            1. Set Redirect type to Permanent (301).
+         1. Click Apply in the right-hand Actions menu.
+         1. Click "Back to Rules"
+         1. Choose the ‘Add Rule’ action from the right pane of the management console, and select the ‘Reverse Proxy Rule’ from the ‘Inbound and Outbound Rules’ category.
+         1. If asked to enable proxy functionality, click “OK”.
+         1. Enter localhost:<Environment port> as where requests will be forwarded
```
Comment (tyliec) [inline src/docs/deployment.md:134]: This is also weird, probably due to the `<` and `>` characters

File: src/docs/deployment.md [hunk 3]
```diff
@@ -1,142 +1,149 @@
...
+         1. Click on enable SSL offloading.
+         1. Click on Use Server Name Indication.
+         1. In Outbound rules, Click on Rewrite domain names
+            - From localhost:<Environment port> to <YourDomain>
```
Comment (tyliec) [inline src/docs/deployment.md:138]: same here

File: src/docs/deployment.md [hunk 4]
```diff
@@ -1,142 +1,149 @@
...
+         1. You should now see a second condition in the list.
+      1. Turn off other features
+         1. Turn off dynamic content compression
+         1. Turn off Output caching - turn off enable cache and enable kernel cache
+      1. Turn on TLS on port 443. This will use [Server Name Indication](https://en.wikipedia.org/wiki/Server_Name_Indication) to use multiple TLS certs, one for each environment:
+         1. Navigate to your website in IIS (left sidebar) and select “Bindings…” on the right hand side.
+         1. Specify https and hostname is <YourDomain>
+         1. Save that
+         1. This page should now show http and https bindings. Close the window.
+      1. Visit http://<YourDomain> from a web browser other than the VM. Your browser should be redirected to https://<YourDomain> and it should use TLS to secure the website.
```
Comment (tyliec) [inline src/docs/deployment.md:149]: same here

File: src/docs/deployment.md [hunk 5]
```diff
@@ -1,142 +1,149 @@
...
+      1. In the next window, select “Default web site”, type “<YourDomainForEnvironment>” in the “Add domains to certificate” field, then click the green Plus sign.
+      1. Click the “Test” button in the top right of the screen and verify that the test succeeds.
+      1. Click the back left arrow button.
+      1. Click the “Request Certificate” button
+      1. Wait until page says “Success” and “Request Completed”.
+      1. Click the back left arrow button if one shows up.
+      1. Close Certify the Web
+1. Open “Internet Information Services (IIS) manager”:
+   1. Do this N times (for each environment you want to create: production, test, etc.) inside of Certify the Web
+      1. Add a reverse proxy in IIS from inbound port 80 to port <Environment port> in the VM and inbound port 443 to port <Environment port> in the VM
```
Comment (tyliec) [inline src/docs/deployment.md:106]: same here

File: src/docs/deployment.md [hunk 6]
```diff
@@ -1,142 +1,149 @@
...
+   1. Download and install Application Request Routing 3.0 https://iis-umbraco.azurewebsites.net/downloads/microsoft/application-request-routing - use x64 version
+1. Download Certify the Web (https://certifytheweb.s3.amazonaws.com/downloads/archive/CertifyTheWebSetup_V5.6.8.exe)
+1. Run Certify the Web installer.
+1. Run Certify the Web once it’s installed.
+   1. Do this N times (for each environment you want to create: production, test, etc.) inside of Certify the Web
+      1. Click “New Certificate”
+      1. Click “OK” when prompted to register a new contact.
+      1. Enter an email address for the HIERR application administrator. Scott’s email is fine.
+      1. Click “Yes, I agree” with the terms for the Let’s Encrypt certificate Authority, then click “Register Contact”.
+      1. In the next window, select “Default web site”, type “<YourDomainForEnvironment>” in the “Add domains to certificate” field, then click the green Plus sign.
```
Comment (tyliec) [inline src/docs/deployment.md:97]: same here

File: src/docs/deployment.md [hunk 7]
```diff
@@ -1,142 +1,149 @@
...
+         ```
+         npm install
+         npx prisma db push
+         npm run build
+         // Linux
+         PORT=<Environment Port> npm run start
+         // Windows
+         npm run start -- --port=<Environment Port>
+         ```
+1. Visit http://localhost:<Environment port> from the same computer and verify that you can see the HIERR application.
```
Comment (tyliec) [inline src/docs/deployment.md:75]: same here

File: src/docs/deployment.md [hunk 8]
```diff
@@ -1,142 +1,149 @@
...
+      1. Add a reverse proxy in IIS from inbound port 80 to port <Environment port> in the VM and inbound port 443 to port <Environment port> in the VM
+         1. Right-click "Sites" on the left under the server and select "Add Website"
+         1. Double click the “URL Rewrite” Icon from the middle pane.
+         1. Click “Add Rule(s)…”
+         1. Select “Blank rule” in the Inbound rules section, then click the OK button.
+         1. Type “HTTPS Redirect” for the name.
+         1. In the Matched URL section:
+            1. Set Requested URL: to Matches the Pattern.
+            1. Set Using to Regular Expressions.
+            1. Enter (.\*) as the Pattern.
```
Comment (tyliec) [inline src/docs/deployment.md:115]: This is rendering as:

Screenshot 2024-05-08 at 10 41 55 PM

File: src/docs/deployment.md (hunk 1 above)
Comment (dinoboy197) [reply src/docs/deployment.md]: I think you found a bunch of old bugs that show up as new due to indentation changes! Fixed anyway :)

File: src/docs/deployment.md (hunk 2 above)
Comment (dinoboy197) [reply src/docs/deployment.md]: Fixed

File: src/docs/deployment.md (hunk 3 above)
Comment (dinoboy197) [reply src/docs/deployment.md]: Fixed

File: src/docs/deployment.md (hunk 4 above)
Comment (dinoboy197) [reply src/docs/deployment.md]: Fixed

File: src/docs/deployment.md (hunk 5 above)
Comment (dinoboy197) [reply src/docs/deployment.md]: Fixed

File: src/docs/deployment.md (hunk 6 above)
Comment (dinoboy197) [reply src/docs/deployment.md]: Fixed

File: src/docs/deployment.md (hunk 7 above)
Comment (dinoboy197) [reply src/docs/deployment.md]: Fixed

File: src/docs/deployment.md (hunk 8 above)
Comment (dinoboy197) [reply src/docs/deployment.md]: Fixed

File: src/docs/deployment.md [hunk 9]
```diff
@@ -1,142 +1,149 @@
...
+         1. In Outbound rules, Click on Rewrite domain names
+            - From localhost:<Environment port> to <YourDomain>
+         1. Click the OK button.
+         1. You should now see a second condition in the list.
+      1. Turn off other features
+         1. Turn off dynamic content compression
+         1. Turn off Output caching - turn off enable cache and enable kernel cache
+      1. Turn on TLS on port 443. This will use [Server Name Indication](https://en.wikipedia.org/wiki/Server_Name_Indication) to use multiple TLS certs, one for each environment:
+         1. Navigate to your website in IIS (left sidebar) and select “Bindings…” on the right hand side.
+         1. Specify https and hostname is <YourDomain>
```
[the remark this answers is not in the document]
Comment (dinoboy197) [reply src/docs/deployment.md]: Fixed

**Review by tyliec — APPROVED** [review]

**Pull request merged**